Remotion/Web/Core/UI/Controls/RendererBase.cs

```diff
@@ -144,13 +143,21 @@ protected IResourceManager GetResourceManager (Type localResourcesType, IResourc
       ArgumentUtility.CheckNotNull ("localResourcesType", localResourcesType);
       ArgumentUtility.CheckNotNull ("controlResourceManager", controlResourceManager);
 
-      var cachedResourceManager = _resourceManagerCache.GetOrAdd (
-          Tuple.Create (localResourcesType, controlResourceManager),
-          key => new Lazy<IResourceManager> (
-              () => ResourceManagerSet.Create (key.Item2, _globalizationService.GetResourceManager (key.Item1)),
-              LazyThreadSafetyMode.ExecutionAndPublication));
+      var table = _resourceManagerCache
+          .GetOrAdd (
+              localResourcesType,
+              _ => new ConditionalWeakTable<IResourceManager, Lazy<IResourceManager>>());
 
-      return cachedResourceManager.Value;
+      if (table.TryGetValue (controlResourceManager, out var cachedResult))
+      {
```

Comment:
As you like: Personally, I drop them for lines like this in favor of a more compact IF structure.